src/amplify-migration/amplifyMigrationScript.ts

```diff
@@ -350,7 +387,9 @@ export async function changeFileContent({
     currentRepoName,
   }));
 
-  const markdownRegex = /\[(.*?)\]\((.*?)\)/g;
+  // we want to be able to modify nested markdown links
+  // eg. [![inline text](/images/someimage.jpg)](/images/somedoc.pdf)
+  const markdownRegex = /(!?\[([^\]]*)\])?\(([^\s]*)\s*(".*")?\)/g;
```

Comment:
might want to break this down into the different sub parts then combine into 1 big regex. this makes the code more maintainable + aids clarity. 

i couldn't really make sense of this even after pasting into `regex101` ._. not a priority if we aren't gonna touch the migration script anytime soon

Reply:
uhm sorry basic qn, how can i break this down into smaller subparts ah? 
maybe why this became extra complicated is because I am trying to capture this specific case:   [![inline text](/images/someimage.jpg)](/images/somedoc.pdf), where there exists inner + outer links. 

while I could do a 
```
const outerLink = `!?\[([^\]]*)\])?\`
const innerLink = `(([^\s]*)\s*(".*")?\`
const markdownRegex = `/(${outerLink}${outerLink})/g`
```

I am not sure if this is more helpful. Is this what you intended?

Reply:
i think in this case, we should be using raw strings and combining them using the `RegExp` constructor - see [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions#creating_a_regular_expression); this is in a similar fashion to what you mentioned - if that works, then it's ok too

Reply:
7ac78b046ff2a4905b318a66b5497b4fabd51bb2 oki